cmd/migration-tool/main.go

```diff
@@ -0,0 +1,524 @@
+
+	if !execute {
+		klog.Infof("Running in Dry-Run mode.  No objects will be created on cluster.")
+	}
+
+	if len(location) > 0 {
+		payloads = readPayloads(location)
+
+		if execute {
+			errors = createPayloads(releasePayloadClient, payloads)
+		}
+	} else {
+		releaseConfigs, err := controller.processImagestreams()
+		if err != nil {
+			klog.Errorf("Unable to gather release configs: %v", err)
+			return
+		}
+		releases, skipped, problems := controller.processReleaseImagestream()
+		payloads = controller.generatePayloads(releases, releaseConfigs)
+
+		klog.Infof("Successfully generated %d ReleasePayload definitions", len(payloads))
+		klog.Warningf("The following %d tag(s) have problems:\n%s", len(problems), strings.Join(problems, ","))
+		klog.Warningf("The following %d tag(s) cannot be migrated:\n%s", len(skipped), strings.Join(skipped, ","))
+
+		if execute {
+			errors = createPayloads(releasePayloadClient, payloads)
+		} else {
+			writePayloads(payloads)
+		}
+	}
+
+	if len(errors) > 0 {
+		klog.Info("The following errors occurred:\n")
+		for _, e := range errors {
+			klog.Warningf(e.Error())
+		}
+	}
+}
+
+type Controller struct {
+	Context       context.Context
+	ImageClient   imageclient.ImageV1Interface
+	Namespace     string
+	Name          string
+	ProwNamespace string
+}
+
+func newController(ctx context.Context, client imageclient.ImageV1Interface, namespace, name, prowNamespace string) *Controller {
+	return &Controller{
+		Context:       ctx,
+		ImageClient:   client,
+		Namespace:     namespace,
+		Name:          name,
+		ProwNamespace: prowNamespace,
+	}
+}
+
+func (c Controller) processImagestreams() (map[string]*releasecontroller.ReleaseConfig, error) {
+	configs := make(map[string]*releasecontroller.ReleaseConfig)
+	list, err := c.ImageClient.ImageStreams(c.Namespace).List(c.Context, metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+	for _, is := range list.Items {
+		var content string
+		var ok bool
+		if content, ok = is.Annotations[releasecontroller.ReleaseAnnotationConfig]; !ok {
+			continue
+		}
+		releaseConfig := &releasecontroller.ReleaseConfig{}
+		if err = json.Unmarshal([]byte(content), &releaseConfig); err != nil {
+			klog.Errorf("Unable to unmarshal release config for %s: %v", is.Name, err)
+			continue
+		}
+		configs[fmt.Sprintf("%s/%s", is.Namespace, is.Name)] = releaseConfig
+	}
+	klog.V(4).Infof("Found %d ReleaseConfigs", len(configs))
+	return configs, nil
+}
+
+type MigrationData struct {
+	Name                       string
+	ReleaseStream              string
+	Phase                      string
+	SourceImageStreamName      string
+	SourceImageStreamNamespace string
+	Status                     releasecontroller.VerificationStatusMap
+	Message                    string
+	Reason                     string
+}
+
+func (c Controller) processReleaseImagestream() ([]MigrationData, []string, []string) {
```

Comment:
to be a controller later, this should take the input as an arg.  `context, imagestreamnamespace, imagestreamname`